### PR TITLE
[backend] Link comparison even on no significant changes

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

File: backend/backend/src/main/java/de/aaaaaaah/velcom/backend/listener/github/GithubPrInteractor.java [hunk 1]
```diff
@@ -571,7 +571,21 @@ private String buildFinishedPrReply(FinishedGithubCommand reply, RunId runId,
 			.append(".");
 
 		if (reasons == null) {
-			builder.append("\nThere were no significant changes.");
+			if (compareToRuns.size() == 1) {  // no idea why we would even have more than 1
```
Comment (Garmelon) [inline backend/backend/src/main/java/de/aaaaaaah/velcom/backend/listener/github/GithubPrInteractor.java:574]: When trying to compare commit `I` against branch `branch` in a scenario like the following:
```
A - B - C - D - E - branch
     \       \
      F - G - H - I
```
compareToRuns would include both B and D.

**Comment by Garmelon** [comment]
Left a comment on the line with `// no idea why we would even have more than 1` explaining when that might happen.

File: backend/backend/src/main/java/de/aaaaaaah/velcom/backend/listener/github/GithubPrInteractor.java (hunk 1 above)
Comment (Kha) [reply backend/backend/src/main/java/de/aaaaaaah/velcom/backend/listener/github/GithubPrInteractor.java:574]: Seems like a stretch to assume that anyone would ever be interested in `B` :sweat_smile: . Anyway, not an issue for Lean.

Comment (Garmelon) [reply backend/backend/src/main/java/de/aaaaaaah/velcom/backend/listener/github/GithubPrInteractor.java:574]: One can easily think up scenarios (though not for Lean's linear structure) where the two or more commits in question are symmetric:
```
      C - D ----.
     /     \     \
A - B       \     G - branch
     \       \   /
      E - F --\-´
           \   H - I
            `-´
```
Comparing `I` against `branch` here would include `D` and `F` in `compareToRuns`.

**Pull request merged**